src/database/entities/Photo.ts

```diff
@@ -48,4 +49,12 @@ export class Photo extends BaseEntity {
   @IsNotEmpty()
   @Column({ type: 'varchar', nullable: false })
   s3Path: string;
+
+  @OneToOne(_ => ColorPalette, {
+    nullable: true,
+    cascadeAll: true,
+    lazy: true,
+  })
+  @JoinColumn()
+  colorPalette: Promise<ColorPalette | null>;
```

Comment:
What is going on here, we're storing Promises in the database?!

Reply:
Of course not 😀, it's because this column is set to `lazy: true` so it's only fetched when the property is accessed and since this happens asynchronously, it has to be wrapped in a promise.